database/src/service/read.rs

```diff
@@ -199,30 +200,45 @@ impl tower::Service<ReadRequest> for DatabaseReadHandle {
 ///
 /// This is the main entrance into all `Request` handler functions.
 /// The basic structure is:
-///
 /// 1. `Request` is mapped to a handler function
 /// 2. Handler function is called
 /// 3. [`Response`] is sent
 fn map_request(
-    _permit: OwnedSemaphorePermit,   // Permit for this request
-    env: Arc<ConcreteEnv>,           // Access to the database
-    request: ReadRequest,            // The request we must fulfill
+    _permit: OwnedSemaphorePermit, // Permit for this request, dropped at end of function
+    env: Arc<ConcreteEnv>,         // Access to the database
+    request: ReadRequest,          // The request we must fulfill
     response_sender: ResponseSender, // The channel we must send the response back to
 ) {
     /* TODO: pre-request handling, run some code for each request? */
-
-    match request {
-        ReadRequest::Example1 => example_handler_1(env, response_sender),
-        ReadRequest::Example2(x) => example_handler_2(env, response_sender, x),
-        ReadRequest::Example3(x) => example_handler_3(env, response_sender, x),
-    }
+    use ReadRequest as R;
+
+    let response = match request {
+        R::BlockExtendedHeader(block) => block_extended_header(&env, block),
+        R::BlockHash(block) => block_hash(&env, block),
+        R::BlockExtendedHeaderInRange(range) => block_extended_header_in_range(&env, range),
+        R::ChainHeight => chain_height(&env),
+        R::GeneratedCoins => generated_coins(&env),
+        R::Outputs(map) => outputs(&env, map),
+        R::NumberOutputsWithAmount(vec) => number_outputs_with_amount(&env, vec),
+        R::CheckKIsNotSpent(set) => check_k_is_not_spent(&env, set),
+        R::BlockBatchInRange(range) => block_batch_in_range(&env, range),
+    };
+
+    // TODO: what do we do if this errors?
```

Comment:
replace by `expect("msg")` if you feel like it. It should never happen

Reply:
no the error should just be ignored, if for example a p2p peer requests some data and then disconnects the oneshot receiver will be dropped and this will panic